taca/analysis/analysis_nanopore.py

```diff
@@ -0,0 +1,278 @@
+    if line_entries[2] == '0':
+        return False
+    else:
+        return True
+
+def get_barcode_kit(sample_sheet):
+    # Figure out which barcode kit was used. Assumes only one kit is ever used.
+    with open(sample_sheet, 'r') as f:
+        for i, line in enumerate(f):
+            if i == 1: # Only need to check first non-header line
+                line_entries = line.split(',')
+    if int(line_entries[2]) <= 12:
+        return 'EXP-NBD104'
+    elif int(line_entries[2]) >= 13:
+        return 'EXP-NBD114'
+    barcode_kit = get_barcode_kit(sample_sheet)
+
+def check_exit_status(status_file):
+    # Read pipeline exit status file and return True if 0, False if anything else
+    with open(status_file, 'r') as f:
+        exit_status = f.readline().strip()
+    if exit_status == '0':
+        return True
+    else:
+        return False
+
+def is_not_transferred(run_id, transfer_log):
+    # Return True if run id not in transfer.tsv, else False
+        with open(transfer_log, 'r') as f:
+            if run_id not in f.read():
+                return True
+            else:
+                return False
+
+def transfer_run(run_dir):
+    #rsync dir to irma
+    logger.info('Transferring run {} to analysis cluster'.format(run_dir))
+    destination = CONFIG.get('nanopore_analysis').get('transfer').get('destination')
+    rsync_opts = {'-Lav': None,
+                  '--no-o' : None,
+                  '--no-g' : None,
+                  '--chmod' : 'g+rw',
+                  '-r' : None,
+                  '--exclude' : 'work'}
+    connection_details = CONFIG.get('nanopore_analysis').get('transfer').get('analysis_server')
+    transfer_object = RsyncAgent(run_dir,
+                                 dest_path=destination,
+                                 remote_host=connection_details['host'],
+                                 remote_user=connection_details['user'],
+                                 validate=False,
+                                 opts=rsync_opts)
+    try:
+        transfer_object.transfer()
+    except RsyncError:
+        logger.warn('An error occurred while transferring {} to the \
+        ananlysis server. Please check the logfiles'.format(run_dir))
+        return False
+    return True
+
+def update_transfer_log(run_id, transfer_log):
+    try:
+        with open(transfer_log, 'a') as f:
+            tsv_writer = csv.writer(f, delimiter='\t')
+            tsv_writer.writerow([run_id, str(datetime.now())])
+    except IOError:
+        logger.warn('Could not update the transfer logfile for run {}. \
+        Please make sure gets updated.'.format(run_id, transfer_log))
+    return
+
+def archive_run(run_dir):
+    # mv dir to nosync
+    logger.info('Archiving run ' + run_dir)
+    archive_dir = CONFIG.get('nanopore_analysis').get('finished_dir')
+    top_dir = '/'.join(run_dir.split('/')[0:-2]) # This is a bit hacky, happy for suggestions on better solutions
```

Comment:
Maybe `pathlib.Path(run_dir).parent.parent`? Not tested, I found it here: https://stackoverflow.com/a/58107778

Reply:
Nice! I'll test it 👍

Reply:
Hm, looks like it might only work for Python > 3.4? We're stuck running 2.7 until we've ported taca-ngi-pipeline and ngi_pipeline. Although, since this will only run on Squiggle I guess I _could_ deploy it in python 3 🤔

Reply:
I think I'll leave it for now but I'll make a note of trying out pathlib once everything has been ported

Reply:
Ah, sorry. I thought we had switched already. I guess it's `os.path` which is the standard package to use. But feel free to keep it like this. Maybe stick in a comment what it does, I had to try it out myself to figure out what the `[0:-2]` did. Once I figured it out, it's obvious though.

Reply:
Done 🙂